File: datasets/jl-keep-code_davinci_001_temp_0.2-0/HumanEval_107_even_odd_palindrome.jl
"""
    Given a positive integer n, return a tuple that has the number of even and odd
    integer palindromes that fall within the range(1, n), inclusive.

    Example 1:

        Input: 3
        Output: (1, 2)
        Explanation:
        Integer palindrome are 1, 2, 3. one of them is even, and two of them are odd.

    Example 2:

        Input: 12
        Output: (4, 6)
        Explanation:
        Integer palindrome are 1, 2, 3, 4, 5, 6, 7, 8, 9, 11. four of them are even, and 6 of them are odd.

    Note:
        1. 1 <= n <= 10^3
        2. returned tuple has the number of even and odd integer palindromes respectively.
    """
function even_odd_palindrome(n::Int64)::Tuple{Int64, Int64} 
    even_count = 0
    odd_count = 0
    for i in 1:n
        if iseven(i)
            even_count += 1
        else
            odd_count += 1
        end
    end
    return even_count, odd_count
end

using Test

@testset begin

candidate = even_odd_palindrome;
	@test(candidate(123) == (8, 13))
	@test(candidate(12) == (4, 6))
	@test(candidate(3) == (1, 2))
	@test(candidate(63) == (6, 8))
	@test(candidate(25) == (5, 6))
	@test(candidate(19) == (4, 6))
	@test(candidate(9) == (4, 5))
	@test(candidate(1) == (0, 1))
end

File: datasets/py-keep-code_davinci_001_temp_0.2/HumanEval_107_even_odd_palindrome.py
from typing import Tuple

def even_odd_palindrome(n: int) -> Tuple[int, int]:
    """
    Given a positive integer n, return a tuple that has the number of even and odd
    integer palindromes that fall within the range(1, n), inclusive.

    Example 1:

        Input: 3
        Output: (1, 2)
        Explanation:
        Integer palindrome are 1, 2, 3. one of them is even, and two of them are odd.

    Example 2:

        Input: 12
        Output: (4, 6)
        Explanation:
        Integer palindrome are 1, 2, 3, 4, 5, 6, 7, 8, 9, 11. four of them are even, and 6 of them are odd.

    Note:
        1. 1 <= n <= 10^3
        2. returned tuple has the number of even and odd integer palindromes respectively.
    """
    even_count = odd_count = 0
    for i in range(1, n + 1):
        if is_palindrome(i):
            if i % 2 == 0:
                even_count += 1
            else:
                odd_count += 1
    return even_count, odd_count


def check(candidate):
    assert candidate(123) == (8, 13)
    assert candidate(12) == (4, 6)
    assert candidate(3) == (1, 2)
    assert candidate(63) == (6, 8)
    assert candidate(25) == (5, 6)
    assert candidate(19) == (4, 6)
    assert candidate(9) == (4, 5)
    assert candidate(1) == (0, 1)

def test_check():
    check(even_odd_palindrome)

test_check()
